INSTALL.md

```diff
@@ -1,3 +1,86 @@
 # Installation
 
 See https://github.com/nerdvegas/rez/wiki/Getting-Started#installation
+
+
+First, install Rez. Download the source, and, from the source directory, run:
+
+```
+]$ python ./install.py
+```
+
+This installs rez to `/opt/rez`. See `install.py -h` for how to install to a
+different location.
+
+Once the installation is complete, a message tells you how to run it:
+
+```
+SUCCESS! To activate Rez, add the following path to $PATH:
+/opt/rez/bin/rez
+
+You may also want to source the completion script (for bash):
+source /opt/rez/completion/complete.sh
+```
+
+Do _not_ move the installation - re-install to a new location if you want to
+change the install path. If you want to install rez for multiple operating
+systems, perform separate installs for each of those systems.
+
+
+# Installation Via Pip
+
+It is possible to install rez with pip, like so:
+
+```
+]$ pip install rez
+```
+
+However, this comes with a caveat - _rez command line tools are disabled once
+inside a rez environment (ie after using the `rez-env` command)_. The reasons
+are given in the next section.
+
+Rez installation via pip is not considered production ready. However, if all you
+want is the rez API, and you don't need its tools to be available within rez
+environments, then you can install with pip.
+
+
+# Why Not Pip For Production?
+
+Rez is not a normal python package. Although it can successfully be installed
+using standard mechanisms such as pip, this comes with a number of caveats.
+Specifically:
+
+* When within a rez environment (ie after using the `rez-env` command), the rez
+  command line tools would not be guaranteed to function correctly;
+* When within a rez environment, other packages' tools (that were also installed
+  with pip) would remain visible, but would not be guaranteed to work.
```

Comment:
This may be addressed using one of the methods you mentioned in #625.

I made a prototype of this a few weeks ago but haven't been testing it in production for more than a few days, and only on Windows. I've updated it to work on Linux just now, you can test it and see if it fits the bill like this:

```bash
$ pip install bleeding-rez
$ rez --version
bleeding-rez-2.33.4
$ rez env python --isolated
```

The environment you'll get is completely void of anything from the parent environment, including any PYTHONPATH or site-packages. Now, this is so barebones, that it also excludes `rez` itself from PATH. So if you wanted it there, you could add it [here](https://github.com/mottosso/bleeding-rez/blob/d99c7baf65dd4b82ab72e80961c0d77aaefa36be/src/rez/cli/_main.py#L152-L157), where `bash` and `lsb_release` are explicitly added as well.

Alternatively, the `REZ_ENVIRONMENT` variable takes a JSON-serialised environment to be used in place of the skeleton environment built into the source. Out of the box you're left with a Docker-like environment, with only the shell itself plus your resolve.